django/db/models/sql/query.py

```diff
@@ -988,6 +988,18 @@ def prepare_lookup_value(self, value, lookups, can_reuse, allow_joins=True):
             pre_joins = self.alias_refcount.copy()
             value = value.resolve_expression(self, reuse=can_reuse, allow_joins=allow_joins)
             used_joins = [k for k, v in self.alias_refcount.items() if v > pre_joins.get(k, 0)]
+        elif isinstance(value, (list, tuple)):
+            processed_values = []
+            used_joins = set()
+            for sub_value in value:
+                if hasattr(sub_value, 'resolve_expression'):
+                    pre_joins = self.alias_refcount.copy()
+                    processed_values.append(sub_value.resolve_expression(
+                        self,
+                        reuse=can_reuse,
+                        allow_joins=allow_joins
+                    ))
+                    used_joins |= set(k for k, v in self.alias_refcount.items() if v > pre_joins.get(k, 0))
```

Comment:
This isn't my area of expertise. @akaariai have you got some time to look at this block in particular?